heat/core/manipulations.py

```diff
@@ -3254,6 +3255,235 @@ def vstack(arrays: Sequence[DNDarray, ...]) -> DNDarray:
+
+    - if 'x.ndim < d', 'x' is promoted to be d-dimensional by prepending new axes.
+    So a shape (3,) array is promoted to (1, 3) for 2-D replication, or shape (1, 1, 3)
+    for 3-D replication (if this is not the desired behavior, promote 'x' to d-dimensions
+    manually before calling this function);
+
+    - if 'x.ndim > d', 'reps' will replicate the last 'd' dimensions of 'x', i.e., if
+    'x.shape' is (2, 3, 4, 5), a 'reps' of (2, 2) will be expanded to (1, 1, 2, 2).
+
+    Parameters
+    ----------
+    x : DNDarray
+        Input
+
+    reps : Sequence[ints,...]
+        Repetitions
+
+    Returns
+    -------
+    tiled : DNDarray
+            Split semantics: if `x` is distributed, the tiled data will be distributed along the
+            same dimension. Note that nominally `tiled.split != x.split` in the case where
+            `len(reps) > x.ndim`.  See example below.
+
+    Examples
+    --------
+    >>> x = ht.arange(12).reshape((4,3)).resplit_(0)
+    >>> x
+    DNDarray([[ 0,  1,  2],
+              [ 3,  4,  5],
+              [ 6,  7,  8],
+              [ 9, 10, 11]], dtype=ht.int32, device=cpu:0, split=0)
+    >>> reps = (1, 2, 2)
+    >>> tiled = ht.tile(x, reps)
+    >>> tiled
+    DNDarray([[[ 0,  1,  2,  0,  1,  2],
+               [ 3,  4,  5,  3,  4,  5],
+               [ 6,  7,  8,  6,  7,  8],
+               [ 9, 10, 11,  9, 10, 11],
+               [ 0,  1,  2,  0,  1,  2],
+               [ 3,  4,  5,  3,  4,  5],
+               [ 6,  7,  8,  6,  7,  8],
+               [ 9, 10, 11,  9, 10, 11]]], dtype=ht.int32, device=cpu:0, split=1)
+    """
+    # check that input is DNDarray
+    sanitation.sanitize_in(x)
+    # check dimensions
+    if x.ndim == 0:
+        x = sanitation.scalar_to_1d(x)
+    # reps to list
+    reps = sanitation.sanitize_sequence(reps)
+    if not all(isinstance(rep, int) for rep in reps):
+        raise TypeError("reps must be a sequence of integers, got {}".format(reps))
+
+    # calculate new gshape, split
+    if len(reps) != x.ndim:
+        added_dims = abs(len(reps) - x.ndim)
+        if len(reps) > x.ndim:
+            new_shape = added_dims * (1,) + x.gshape
+            new_split = None if x.split is None else x.split + added_dims
+            x = x.reshape(new_shape, axis=new_split)
+        else:
+            reps = added_dims * [1] + reps
+    split = x.split
+    # "t_" indicates process-local torch tensors
+    t_x = x.larray
+    out_gshape = tuple(
+        (torch.tensor(x.shape, device=t_x.device) * torch.tensor(reps, device=t_x.device)).tolist()
+    )
+    if split is None or reps[split] == 1:
+        # no repeats along the split axis: local operation
+        t_tiled = t_x.repeat(reps)
+        return DNDarray(
+            t_tiled,
+            out_gshape,
+            dtype=x.dtype,
+            split=split,
+            device=x.device,
+            comm=x.comm,
+            balanced=x.balanced,
+        )
+    else:
```

Comment:
very very minor comment: since you have a return in the line above, you dont need an else block here at all